My Project/clsTranslateWinForms.vb

```diff
@@ -1 +1,638 @@
+            Return ""
+        End If
+
+        Dim dctComponents As Dictionary(Of String, Component) = New Dictionary(Of String, Component)
+        GetDctComponentsFromMenuItems(clsMenuItems, dctComponents)
+
+        Dim strMenuItemsAsCsv As String = ""
+        For Each clsComponent In dctComponents
+
+            If TypeOf clsComponent.Value Is ToolStripItem Then
+                Dim clsMenuItem As ToolStripItem = DirectCast(clsComponent.Value, ToolStripItem)
+                strMenuItemsAsCsv &= clsControl.Name & "," & clsComponent.Key & "," & GetCsvText(clsMenuItem.Text) & vbCrLf
+            Else
+                MsgBox("Developer Error: Translation dictionary entry (" & clsControl.Name & "," & clsComponent.Key & ") contained unexpected value type.")
+            End If
+
+        Next
+        Return strMenuItemsAsCsv
+    End Function
+
+    '''--------------------------------------------------------------------------------------------
+    ''' <summary>
+    '''    Populates dictionary <paramref name="dctComponents"/> with the control 
+    '''    <paramref name="clsControl"/> and its children.    
+    '''    The dictionary can then be used to conveniently translate the control text (see other
+    '''    functions and subs in this class).
+    ''' </summary>
+    '''
+    ''' <param name="clsControl">       The control used to populate the dictionary. </param>
+    ''' <param name="dctComponents">    [in,out] Dictionary to store the control and its children. 
+    '''                                 </param>
+    '''--------------------------------------------------------------------------------------------
+    Private Shared Sub GetDctComponentsFromControl(clsControl As Control,
+                                                   ByRef dctComponents As Dictionary(Of String, Component),
+                                                   Optional strParentName As String = "")
+        If IsNothing(clsControl) OrElse IsNothing(clsControl.Controls) OrElse IsNothing(dctComponents) Then
+            Exit Sub
+        End If
+
+        'if control is valid, then add it to the dictionary
+        Dim strControlName As String = ""
+        If Not String.IsNullOrEmpty(clsControl.Name) Then
+            strControlName = If(String.IsNullOrEmpty(strParentName), clsControl.Name, strParentName & "_" & clsControl.Name)
+            If Not dctComponents.ContainsKey(strControlName) Then  'ignore components that are already in the dictionary
+                dctComponents.Add(strControlName, clsControl)
+            End If
+        End If
+
+        For Each ctlChild As Control In clsControl.Controls
+
+            'Recursively process different types of menus and child controls
+            If TypeOf ctlChild Is MenuStrip Then
+                Dim clsMenuStrip As MenuStrip = DirectCast(ctlChild, MenuStrip)
+                GetDctComponentsFromMenuItems(clsMenuStrip.Items, dctComponents)
```

Comment:
@lloyddewit line 277 could be simplified to.

`GetDctComponentsFromMenuItems(DirectCast(ctlChild, ToolStrip).Items, dctComponents)`

Same applies to line 280

Reply:
agreed, done